### PR TITLE
MICN-113  Switch to filtered case note type and sub type endpoint

### PR DESCRIPTION
**Pull request by bill-moj**
This PR amends `Case Note Type` _types_ to better reflect their usages, and set filters on `GET /case-notes/types` explicitly:

1. separate CaseNoteType and CaseNoteSubType, so that the parent type and child type no longer share the same type
2. remove unnecessary property on CaseNoteType (parent type)
3. replace `GET /case-notes/types-for-user` call with `GET /case-notes/types` using the correct filters. `GET /case-notes/type-for-user` endpoint will be deprecated.
4. explicitly set query params of what kind of CaseNoteSubTypes to be included for `GET /case-notes/types` calls (namely, selectable by DPS_USER or by ALL; include INACTIVE and/or RESTRICTED subtypes), based on the user's roles.

### CODE REVIEW

**Review by iterative-it — DISMISSED** [review]
LGTM

File: server/data/interfaces/caseNotesApi/CaseNoteType.ts [hunk 1]
```diff
@@ -9,12 +9,11 @@ export default interface CaseNoteType {
...
-  includeSensitive?: boolean
-  includeRestrictedUse?: boolean
+  includeRestricted?: boolean
   includeInactive?: boolean
 }
 
 export interface CaseNotesTypeQueryParams extends QueryParams {
   selectableBy: 'ALL' | 'DPS_USER'
-  include: ('INACTIVE' | 'SENSITIVE' | 'RESTRICTED')[]
+  include: ('INACTIVE' | 'RESTRICTED')[]
```
Comment (iterative-it) [inline server/data/interfaces/caseNotesApi/CaseNoteType.ts:18]: The include query parameter has been removed from the API

Comment (bill-moj) [reply server/data/interfaces/caseNotesApi/CaseNoteType.ts]: fixed

File: server/services/caseNotesService.ts [hunk 2]
```diff
@@ -64,7 +66,11 @@ export default class CaseNotesService {
     const errors: HmppsError[] = validateDateRange(queryParams.startDate, queryParams.endDate)
 
     let pagedCaseNotes: PagedList<CaseNotePageData>
-    const caseNoteTypes = await caseNotesApiClient.getCaseNoteTypes()
+    const caseNoteTypes = await caseNotesApiClient.getCaseNoteTypes({
+      dpsUserSelectableOnly: false,
+      includeInactive: true,
+      includeRestricted: canEditSensitiveCaseNotes,
```
[the remark this answers is not in the document]
Comment (bill-moj) [reply server/services/caseNotesService.ts]: fixed

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**